.github/workflows/cspell.yml

```diff
@@ -11,8 +11,14 @@ env:
   FORCE_COLOR: 2
   NODE: 16
 
+permissions:
+  contents: read
+
 jobs:
   cspell:
+    permissions:
+      contents: read  # for streetsidesoftware/cspell-action to fetch files for commit
```

Comment:
I wonder, why is this needed since you have specified the same globally?

Reply:
Hi @XhmikosR this is needed because there are two permissions specified for that job at the job level. When permissions are specified at the job level, only those listed in the job are provided to the token [1]. So if `contents: read` is not listed again, the job will only get `pull-requests: read`. 

If no permissions are listed at the job level, it gets the permissions from the workflow level. If the job only needed `contents: read`, we could have omitted the permissions section at the job level altogether. 

Please let me know if you have follow up questions. Thanks!

```
permissions:
      contents: read  # for streetsidesoftware/cspell-action to fetch files for commit
      pull-requests: read  # for streetsidesoftware/cspell-action to fetch commits for PR
```

[1] When the permissions key is used, all unspecified permissions are set to no access, with the exception of the metadata scope, which always gets read access. https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token

Reply:
@XhmikosR any update on the pull request review? Please let me know if you have any follow up questions.